src/popup/modules/InitQrCode.js

```diff
@@ -5,12 +5,13 @@ import * as MessageHandler from "/common/modules/MessageHandler.js";
 import * as QrCreator from "./QrCreator.js";
 import * as BrowserCommunication from "./BrowserCommunication.js";
 import * as UserInterface from "./UserInterface.js";
+import { retryPromise } from "../../common/modules/HelperFunctions.js";
 
 /* globals */
 export let initCompleted = false;
 
 // init modules
-const queryBrowserTabs = browser.tabs.query({active: true, currentWindow: true});
+const queryBrowserTabs = () => browser.tabs.query({active: true, currentWindow: true});
```

Comment:
Actually, with your change, this now does not need to be a the global scope anymore. This variable is only used once, so you do not need to save it in a global var. :smile:

Reply:
True that, I moved it inside the function